Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java

```diff
@@ -1656,6 +1677,69 @@ protected Bitmap getMarkerBitmapFromView(String locationName, String imageName,
         return returnedBitmap;
     }
 
+    protected void onMapUpdate() {
+        try {
+            MapRemoteConfig mapRemoteConfig = getMapRemoteConfig();
+            if(mapRemoteConfig.enableMapRecenter) {
+                String merchant = bridgeComponents.getContext().getResources().getString(in.juspay.mobility.app.R.string.service);
+                String merchantType = merchant.contains("partner") || merchant.contains("driver") || merchant.contains("provider") ? "DRIVER" : "USER";
+                if (merchantType.equals("USER")) {
+                    ExecutorManager.runOnMainThread(() -> {
+                        try {
+                            if (!isIdleListenerActive) {
+                                googleMap.setOnCameraIdleListener(() -> {
+                                    mapRecenterHandler.removeCallbacksAndMessages(null);
+                                    mapRecenterHandler.postDelayed(() -> {
+                                        if(isGestureMovement) isMapMoved = true;
+                                        isMapIdle = true;
+                                    }, mapRemoteConfig.recenterDelay);
+                                });
+                                isIdleListenerActive = true;
+                            }
+                            if (!isMoveListenerActive) {
+                                googleMap.setOnCameraMoveStartedListener(reason -> {
+                                    if (reason == GoogleMap.OnCameraMoveStartedListener.REASON_GESTURE) {
+                                        isGestureMovement = true;
+                                    } else {
+                                        isGestureMovement = false;
+                                    }
+                                    mapRecenterHandler.removeCallbacksAndMessages(null);
+                                    isMapIdle = false;
+                                });
+                                isMoveListenerActive = true;
+                            }
+                        } catch (Exception e) {
+                            Log.e(LOG_TAG, "Error in onMapUpdate Executor " + e);
+                        }
+                    });
+                }
+            }
+        } catch (Exception e) {
+            Log.e(LOG_TAG, "Error in onMapUpdate ", e);
+        }
+    }
+
+    private void removeOnMapUpdate() {
```

Comment:
make these two functions a JBridge too. and add callback strings which will be helpful in future.